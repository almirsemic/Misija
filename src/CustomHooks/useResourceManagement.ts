const useResourceManagement = () => {
  const addResource = async (newResource: any) => {
    try {
      return { success: true, message: 'Resource added successfully.' }
    } catch (error) {
      return { success: false, message: 'Failed to add resource.' }
    }
  }

  const editResource = async (resourceId: string, updatedResource: any) => {
    try {
      return { success: true, message: 'Resource updated successfully.' }
    } catch (error) {
      return { success: false, message: 'Failed to update resource.' }
    }
  }

  const deleteResource = async (resourceId: string) => {
    try {
      return { success: true, message: 'Resource deleted successfully.' }
    } catch (error) {
      return { success: false, message: 'Failed to delete resource.' }
    }
  }

  return { addResource, editResource, deleteResource }
}

export default useResourceManagement
