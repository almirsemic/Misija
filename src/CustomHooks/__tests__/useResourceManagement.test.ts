import useResourceManagement from "../useResourceManagement";


describe('useResourceManagement', () => {
  it('should add a resource when user is logged in', () => {
    const hook = useResourceManagement();
    const newResource = { name: 'New Resource' };
    
    const result = hook.addResource(newResource);
    
    expect(result).toMatchSnapshot();
  });

  it('should return error message when user is not logged in while adding a resource', () => {
    const hook = useResourceManagement();
    const newResource = { name: 'New Resource' };
    
    const result = hook.addResource(newResource);

    expect(result).toMatchSnapshot();
  });

  it('should edit a resource when user is logged in', () => {
    const hook = useResourceManagement();
    const resourceId = '123';
    const updatedResource = { name: 'Updated Resource' };
    
    const result = hook.editResource(resourceId, updatedResource);

    expect(result).toMatchSnapshot();
  });

  it('should return error message when user is not logged in while editing a resource', () => {
    const hook = useResourceManagement();
    const resourceId = '123';
    const updatedResource = { name: 'Updated Resource' };
    
    const result = hook.editResource(resourceId, updatedResource);
    
    expect(result).toMatchSnapshot();
  });

  it('should delete a resource when user is logged in', () => {
    const hook = useResourceManagement();
    const resourceId = '123';
    
    const result = hook.deleteResource(resourceId);

    expect(result).toMatchSnapshot();
  });

  it('should return error message when user is not logged in while deleting a resource', () => {
    const hook = useResourceManagement();
    const resourceId = '123';
    
    const result = hook.deleteResource(resourceId);

    expect(result).toMatchSnapshot();
  });
});