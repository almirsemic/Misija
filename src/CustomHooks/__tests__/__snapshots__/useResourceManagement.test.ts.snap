// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useResourceManagement should add a resource when user is logged in 1`] = `Promise {}`;

exports[`useResourceManagement should delete a resource when user is logged in 1`] = `Promise {}`;

exports[`useResourceManagement should edit a resource when user is logged in 1`] = `Promise {}`;

exports[`useResourceManagement should return error message when user is not logged in while adding a resource 1`] = `Promise {}`;

exports[`useResourceManagement should return error message when user is not logged in while deleting a resource 1`] = `Promise {}`;

exports[`useResourceManagement should return error message when user is not logged in while editing a resource 1`] = `Promise {}`;
